app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -2460,27 +2484,51 @@ private void setMuteButton(@NonNull final ImageButton button, final boolean isMu
     //////////////////////////////////////////////////////////////////////////*/
     //region ExoPlayer listeners (that didn't fit in other categories)
 
+    /**
+     * <p>Listens for event or state changes on ExoPlayer. When any event happens, we check for
+     * changes in the currently-playing metadata and update the encapsulating
+     * {@link Player}. Downstream listeners are also informed.</p>
+     *
+     * <p>When the renewed metadata contains any error, it is reported as a notification.
+     * This is done because not all source resolution errors are {@link PlaybackException}, which
+     * are also captured by {@link ExoPlayer} and stops the playback.</p>
+     *
+     * @param player The {@link com.google.android.exoplayer2.Player} whose state changed.
+     * @param events The {@link com.google.android.exoplayer2.Player.Events} that has triggered
+     *               the player state changes.
+     **/
     @Override
-    public void onTimelineChanged(@NonNull final Timeline timeline, final int reason) {
-        if (DEBUG) {
-            Log.d(TAG, "ExoPlayer - onTimelineChanged() called with "
-                    + "timeline size = [" + timeline.getWindowCount() + "], "
-                    + "reason = [" + reason + "]");
-        }
-
-        maybeUpdateCurrentMetadata();
-        // force recreate notification to ensure seek bar is shown when preparation finishes
-        NotificationUtil.getInstance().createNotificationIfNeededAndUpdate(this, true);
```

Comment:
Why was this removed @karyogamy?

Reply:
Updates still happen on timeline changes. This method, and other calls to `maybeUpdateCurrentMetadata` are now conslidated into `onEvents` which is triggered on any ExoPlayer change instead of scattering them into different callbacks.